src/engines/aurora/gui.h

```diff
@@ -115,10 +115,14 @@ class GUI : boost::noncopyable {
 	/** Callback that's triggered when a widget was activated. */
 	virtual void callbackActive(Widget &widget);
 
+	void addChild(GUI*);
```

Comment:
Also, you should probably add comments that these two methods should not be called in any of the GUI event handlers (because that will mess with the iterator in run()). And that the child GUIs will not have their ownership transferred to the parent GUI (i.e. the child GUIs still have to be deallocated as normal).